packages/compass-aggregations/src/modules/pipeline-builder/pipeline-builder.ts

```diff
@@ -23,13 +27,32 @@ export class PipelineBuilder {
     this.sourceToStages();
   }
 
+  set source(source: string) {
+    this._source = source;
+    this.parseSourceToPipeline();
+  }
+
+  get source() {
+    return this._source;
+  }
+
+  private parseSourceToPipeline() {
+    try {
+      this.pipeline = parseEJSON(this.source, { mode: ParseMode.Loose });
+    } catch (e) {
+      // After setting source, its either validated or mapped to stages
+      // where the whole parsing takes places and errors are handled.
+    }
```

Comment:
This really should be set to something meaningful like null when parsing fails, otherwise you end up with pipeline builder state being not in sync